client/protocols/trace.py

```diff
@@ -0,0 +1,125 @@
+"""Implements the TraceProtocol, which traces code and
+provides a Python Tutor visualization.
+"""
+
+import datetime as dt
+import logging
+import json
+
+from pytutor import generate_trace
+from pytutor import server
+
+from client.protocols.common import models
+from client.sources.doctest import models as doctest_models
+from client.utils import format
+log = logging.getLogger(__name__)
+
+class TraceProtocol(models.Protocol):
+    """ Trace a specific test and render a JSON. """
+
+    def run(self, messages, env=None):
+        """Run gradeable tests and print results and return analytics.
+        """
+        if not self.args.trace:
+            return
+        tests = self.assignment.specified_tests
+        messages['tracing'] = {
+            'begin': get_time(),
+        }
+        if not self.args.question:
+            with format.block('*'):
+                print("Could not trace: Please specify a question to trace.")
+                print("Example: python3 ok --trace -q <name>")
+            return
+
+        test = tests[0]
+        data = test.get_code()
```

Comment:
Should probably test that your test case has a get_code attr (since it's not actually part of the test base model) - we might want to add it there - (and just have it return `{}` - or throw an exception and catch it here)